### DISCUSSION TITLE
Call configuration `resolve` from outside

### DISCUSSION THREAD

**Discussion by csvtuda** [badeball/cypress-cucumber-preprocessor#1056] [Q&A] [answered]
Hey,

I'm maintaining a plugin which builds on the `cypress-cucumber-preprocessor`. In particular, I'd like to reuse the JSON report it generates to upload it to Xray (test management plugin for Jira). To do this I'd need to:
- verify that `json.enabled` is `true`
- retrieve the `json.output` property from the preprocessor's configuration

Because these properties aren't just set in one location and can even be configured on the fly using environment variables, I really do not want to rebuild another half-baked configuration parser just for accessing these two properties.

Therefore I was wondering if it is safe to call `resolve` from outside/my plugin just like you're doing here: https://github.com/badeball/cypress-cucumber-preprocessor/blob/95caf826b3a26f53301c10e05054c6207e223702/lib/plugin-event-handlers.ts#L190
Are there any side effects I'd need to be aware of?

Currently, calling it immediately provides access to the complete configuration which I can then keep on using.

**Comment by badeball** [comment]
Hi @csvtuda, that should be fine. The method doesn't have any side effects. Are you able to retrieve it though, given my module uses conditional exports?

**Reply by csvtuda** [reply]
Thanks @badeball, that's good to know. Yeah, the import works but is quite ugly: 

```ts
import { resolve } from "node_modules/@badeball/cypress-cucumber-preprocessor/dist/preprocessor-configuration";

export default defineConfig({
  e2e: {
    async setupNodeEvents(on, config) {
      const x = await resolve(config, config.env, "/");
      console.log(x);
      return config;
    },
    // ...
  },
});
```

I have yet to really integrate it in all workflows, simple experiments like above worked so far. But I don't know if other bundlers or interpreter settings might lead to problems down the line.

Obviously I'd love to have a way to access the parsed configuration, but if you're not feeling like this would be worth it I'll happily accept the answer too.

**Comment by badeball** [comment, marked as answer]
> Yeah, the import works but is quite ugly:

Definitely don't do that, it will surely cause problems.

I noticed however that I do actually export `resolve(..)`, albeit as [`resolvePreprocessorConfiguration`](https://github.com/badeball/cypress-cucumber-preprocessor/blob/v18.0.1/lib/entrypoint-node.ts#L10).

```js
import { resolvePreprocessorConfiguration } from "@badeball/cypress-cucumber-preprocessor";

// ...
```

**Reply by csvtuda** [reply]
Works like a charm, awesome! Thank you very much.
